Assets/Scripts/Gamestate/PlayerManager.cs

```diff
@@ -191,7 +191,8 @@ private void OnDeath(HealthController healthController, float damage, DamageInfo
         {
             killer = lastPlayerThatHitMe;
         }
-        onDeath?.Invoke(killer, this, info);
+        if (isAlive)
+            onDeath?.Invoke(killer, this, info);
```

Comment:
Note that by guarding only this invocation, you're not preventing the ridiculous amount of knockback (and multiple invocations of the spectator camera code) that the shotgun can cause.

Reply:
I agree that it shoudl return itself, but the knockback stems from somewhere 
different (it also happens when you shoot an explosive barrel with a shotgun and get more knockback while still living)

Reply:
Oh dear.